externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/operation/SetCurrentCatalog.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.trino.operation
+
+import com.google.common.collect.ImmutableList
+import io.trino.client.{ClientSession, ClientTypeSignature, ClientTypeSignatureParameter, Column}
+import io.trino.client.ClientStandardTypes.VARCHAR
+import io.trino.client.ClientTypeSignature.VARCHAR_UNBOUNDED_LENGTH
+
+import org.apache.kyuubi.operation.{IterableFetchIterator, OperationType}
+import org.apache.kyuubi.session.Session
+
+class SetCurrentCatalog(session: Session, catalog: String)
+  extends TrinoOperation(OperationType.EXECUTE_STATEMENT, session) {
+
+  override protected def runInternal(): Unit = {
+    try {
+      val session = trinoContext.clientSession.get
+      var builder = ClientSession.builder(session)
+      builder = builder.withCatalog(catalog)
+      trinoContext.clientSession.set(builder.build())
+
+      val clientTypeSignature = new ClientTypeSignature(
+        VARCHAR,
+        ImmutableList.of(ClientTypeSignatureParameter.ofLong(VARCHAR_UNBOUNDED_LENGTH)))
+      val column = new Column("TABLE_CAT", VARCHAR, clientTypeSignature)
```

Comment:
this is incorrect, the result schema is not TABLE_CAT

Reply:
How about setting schema "output" in set catalog/db schema? Or set to no resultset, similar to Hive.

Reply:
better empty